### PR TITLE
Update motr-design-doc-list.rst

### PR DESCRIPTION
**Pull request by VenkyOS**

### CODE REVIEW

**Comment by huanghua78** [comment]
Please update the comment, list changes in this PR.

**Comment by VenkyOS** [comment]
@huanghua78 - There were some errors in the table that I corrected.

File: doc/motr-design-doc-list.rst [hunk 1]
```diff
@@ -247,17 +247,17 @@ Motr - Design Document List
...
+|       |                                    |        |`HLD of SNS Client </doc/PDF/HLD_of_SNS_client.pdf>`_                                            |   `<RPC_Layer_Core.rst>`_                                                                     |High level design of an SNS client module for C2 T1                                                                                                                                                                                                       |`High level design of an SNS client module for C2 T1 <https://seagatetechnology.sharepoint.com/:w:/r/sites/gteamdrv1/tdrive1224/_layouts/15/Doc.aspx?sourcedoc=%7B4A22CF7B-82D0-49AE-B355-CB777A836472%7D&file=HLD%20of%20SNS%20client.docx&action=default&mobileredirect=true>`_                                      |
```
Comment (huanghua78) [inline doc/motr-design-doc-list.rst:260]: I think you made a wrong link here.

Comment (VenkyOS) [reply doc/motr-design-doc-list.rst]: Incorporated.

**Review by huanghua78 — APPROVED** [review]
Looks fine.

**Pull request merged**